tests/test_flow/test_persistence_compatibility.py

```diff
@@ -0,0 +1,34 @@
+import pytest
+import shutil
+
+from .generate_test_compatibility_cache import Harness, CACHE_TEST_DIR
+
+
+@pytest.fixture
+def older_serialized_cache_harness(tmp_path):
+    # shutil.copytree dest should not exist
+    tmp_cache_path = tmp_path.joinpath('test_cache')
+    shutil.copytree(CACHE_TEST_DIR, tmp_cache_path)
+    harness = Harness(tmp_cache_path)
+    return harness
+
+
+# Tests caching backwards compatibility by loading and deserializaing
+# an old snapshot of the cache. Test failure indicates that the changes
+# made to the caching layer is backwards incompatible.
+# In case of a failure, either
+# a) fix the caching logic so it's backwards compatible or
+# b) update the cache schema version and generate a new cache snapshot.
```

Comment:
We should probably point out exactly where and how to update the cache schema version.